src/core/assistant-message/kilocode/native-tool-call.ts

```diff
@@ -57,3 +57,24 @@ export function parseDoubleEncodedParams(obj: any): any {
 	// Primitive types (number, boolean, etc.) return as-is
 	return obj
 }
+
+const knownWrapperPropertyNames = [
+	"arguments", // seen with qwen3-coder
+	"result", // seen with gemini-2.5-flash, haiku-4.5, sonnet-4.5
+]
+
+export function removeWrapperObject(toolName: string, parsedArgs: Record<string, unknown>, nestingLevel: number) {
```

Comment:
I noticed that the actual arguments are often wrapped in objects with a single property, sometimes multiple levels deep.

There are probably less hacky ways to do this, I was thinking of defining the parameters as zod4 schemas so we can look for an object that matches the schema.